### PR TITLE
add Create TSavingAccount and Update MCif

### PR DESCRIPTION
**Pull request by 261001**
Cuplikan layar 2024-12-16 153331
Cuplikan layar 2024-12-16 201144

Request (Put/update)
{
"customerName": "andis",
"dateOfBirth": "2007-12-14",
"address": "dago",
"phoneNumber": "09857575758787",
"email": "andis@gmail.com",
"idNumber": "7439779799777",
"idNumberType": "1"
}
Response (Put/Update)
{
    "createdBy": null,
    "createdAt": null,
    "updateBy": null,
    "updateAt": null,
    "cifId": "4028e49293c09c600193c09cdf050000",
    "customerName": "andis",
    "dateOfBirth": "2007-12-14",
    "address": "dago",
    "phoneNumber": "09857575758787",
    "email": "andis@gmail.com",
    "idNumber": "7439779799777",
    "rNumberType": {
        "typeId": "1",
        "typeCode": "0101",
        "typeName": "KTP"
    },
    "rStatus": {
        "statusId": "1",
        "statusCode": "001",
        "statusName": "Pending"
    },
    "mUserAuthorizationBy": {
        "createdBy": null,
        "createdAt": null,
        "updateBy": null,
        "updateAt": null,
        "userId": "12345",
        "isDeleted": false,
        "email": "fauziindrawan219@gmail.com",
        "userName": "fauzi",
        "password": null,
        "fullname": "fauzi ganteng",
        "token": null,
        "sessionId": null,
        "muserRole": {
            "userRoleId": "1",
            "roleCode": "1919",
            "roleName": "superAdmin",
            "deleted": false
        }
    },
    "authorization_at": null,
    "deleted": false
}

### CODE REVIEW

**Comment by xuibdg** [comment]
tambahkan http endpoint, dan request response nya disini

File: src/main/java/com/b2camp/simple_core_banking/SimpleCoreBankingApplication.java [hunk 1]
```diff
@@ -2,8 +2,10 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
 
 @SpringBootApplication
+@ComponentScan(basePackages = "com.b2camp.simple_core_banking")
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/SimpleCoreBankingApplication.java]: Do you know the function about @ComponentScan? if you know then delete @ComponentScan

File: src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java [hunk 2]
```diff
@@ -1,18 +1,38 @@
...
+    private MCifRepository mCifRepository;
+
     MCifController(MCifService mCifService) {
         this.mCifService = mCifService;
     }
 
+    @PutMapping("/{id}")
+    public ResponseEntity<?> updateCif(@PathVariable String id, @RequestBody MCif mCifEntity) {
+        MCif existingUser = mCifRepository.findById(id).orElse(null);
+        if (existingUser != null) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/controller/MCifController.java]: why is there logic in the controller? please explain this is logic

File: src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java [hunk 3]
```diff
@@ -159,4 +160,7 @@ public String getStatusName() {
     public void setStatusName(String statusName) {
         this.statusName = statusName;
     }
+
+    public void getAuthorizationAt(Timestamp authorizationAt) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/dto/MCifResponse.java]: why is there a param in gatter?

File: src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountRequest.java [hunk 4]
```diff
@@ -13,19 +14,22 @@ public void setAccountNumber(String accountNumber) {
         this.accountNumber = accountNumber;
     }
 
+    //getter setter untuk cifID
     public String getCifId() {
-        return cifId;
+        return cifId = cifId;
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/dto/TSavingAccountRequest.java]: why cifId = cifId, are the values ​​of the two different?

File: src/main/java/com/b2camp/simple_core_banking/entity/TSavingAccount.java [hunk 5]
```diff
@@ -163,11 +152,36 @@ public void setAuthorizationAt(LocalDateTime authorizationAt) {
         this.authorizationAt = authorizationAt;
     }
 
+
     public String getSavingId() {
         return savingId;
     }
 
     public void setSavungId(String savingId) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/entity/TSavingAccount.java]: what is savung?

File: src/main/java/com/b2camp/simple_core_banking/service/MCifService.java [hunk 6]
```diff
@@ -1,4 +1,10 @@
 package com.b2camp.simple_core_banking.service;
 
+import com.b2camp.simple_core_banking.dto.MCifRequest;
+import com.b2camp.simple_core_banking.dto.MCifResponse;
+import org.springframework.stereotype.Service;
+
+@Service
 public interface MCifService {
+    MCifResponse updateCif(String McifID, MCifRequest request);
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/MCifService.java]: rename param to cifId

File: src/main/java/com/b2camp/simple_core_banking/service/TSavingAccountService.java [hunk 7]
```diff
@@ -1,4 +1,14 @@
...
+import com.b2camp.simple_core_banking.dto.TSavingAccountRequest;
+import com.b2camp.simple_core_banking.dto.TSavingAccountResponse;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
 public interface TSavingAccountService {
+
+    TSavingAccountResponse createe(TSavingAccountRequest tSavingAccountRequest);
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/TSavingAccountService.java]: rename method to create

File: src/main/java/com/b2camp/simple_core_banking/service/impl/MCifServiceImpl.java [hunk 8]
```diff
@@ -1,7 +1,70 @@
...
+    @Override
+    public MCifResponse updateCif(String McifID, MCifRequest request) {
+        MCif mCif = mCifRepository.findById(McifID).orElse(null);
+        mCifRepository.save(mCif);
+        buildToEntityUpdate(mCif, request);
+
+        return buildToResponseUpdate(mCif);
+    }
+
+    private MCifResponse buildToResponseUpdate(MCif mCif) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/impl/MCifServiceImpl.java]: rename method to buildToResponseForUpdate

File: src/main/java/com/b2camp/simple_core_banking/service/impl/MCifServiceImpl.java [hunk 9]
```diff
@@ -1,7 +1,70 @@
...
+        response.setIdNumber(mCif.getIdNumber());
+        response.setPhoneNumber(mCif.getPhoneNumber());
+        response.setIdNumberType(mCif.getIdNumber());
+        response.setStatusId(mCif.getrStatus().getStatusId());
+        response.getAuthorizationAt(mCif.getAuthorization_at());
+        response.setDeleted(false);
+        return response;
+    }
+
+    private void buildToEntityUpdate(MCif mCif, MCifRequest request) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/impl/MCifServiceImpl.java]: rename method to buildToEntityForUpdate

File: src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java [hunk 10]
```diff
@@ -1,6 +1,61 @@
...
+    @Autowired
+    private TSavingAccountRepository tSavingAccountRepository;
+
+    @Autowired
+    private RStatusRepository rStatusRepository;
+
+
+    @Transactional
+    @Override
+    public TSavingAccountResponse createe(TSavingAccountRequest tSavingAccountRequest) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java]: rename method to create

File: src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java [hunk 11]
```diff
@@ -1,6 +1,61 @@
...
+        TSavingAccount tSavingAccount = new TSavingAccount();
+        buildToEntityy(tSavingAccount, tSavingAccountRequest);
+        tSavingAccountRepository.save(tSavingAccount);
+        return buildToResponsee(tSavingAccount);
+    }
+
+    private void buildToEntityy(TSavingAccount tSavingAccount, TSavingAccountRequest request) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java]: rename to buildToEntity

File: src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java [hunk 12]
```diff
@@ -1,6 +1,61 @@
...
+        tSavingAccount.setSavingId(request.getSavingId());
+        tSavingAccount.setBeginBalance(new BigDecimal("1000.00"));
+        tSavingAccount.setEndBalance(new BigDecimal("1000.00"));
+        tSavingAccount.setCurrentBalance(new BigDecimal("1000.00"));
+        tSavingAccount.setIsDeleted(false);
+        tSavingAccount.setAuthorizationAt(Timestamp.from(Instant.now()).toLocalDateTime());
+
+    }
+
+    private TSavingAccountResponse buildToResponsee(TSavingAccount tSavingAccount) {
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java]: rename to buildToEntity

File: src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java [hunk 13]
```diff
@@ -1,6 +1,61 @@
...
+        tSavingAccountRepository.save(tSavingAccount);
+        return buildToResponsee(tSavingAccount);
+    }
+
+    private void buildToEntityy(TSavingAccount tSavingAccount, TSavingAccountRequest request) {
+        tSavingAccount.setAccountNumber(request.getAccountNumber());
+        tSavingAccount.setSavingId(request.getSavingId());
+        tSavingAccount.setBeginBalance(new BigDecimal("1000.00"));
+        tSavingAccount.setEndBalance(new BigDecimal("1000.00"));
+        tSavingAccount.setCurrentBalance(new BigDecimal("1000.00"));
```
Comment (xuibdg) [inline src/main/java/com/b2camp/simple_core_banking/service/impl/TSavingAccountServiceImpl.java]: BeginBalance, EndBalance and CurrentBalance, Why is the setter code not based on request?

**Review by Jundillah123 — APPROVED** [review]

**Review by Yaksyin — APPROVED** [review]

**Review by Yaksyin — APPROVED** [review]

**Review by xuibdg — APPROVED** [review]

**Pull request merged**